### ISSUE TITLE
Example - failover: error when deploying using sample_failover.yaml

### ISSUE THREAD

**Issue by curtkersey** [F5Networks/f5-google-gdm-templates-v2#7]
## Describe the bug
Deploying the sample_failover.yaml example, and it is not working.  There is an error in /var/log/cloud/bigIpRuntimeInit.log:

2022-03-22T13:21:57.600Z [19211]: info: Configuration file: /config/cloud/runtime-init-conf.yaml
2022-03-22T13:21:57.626Z [19211]: info: Validating provided declaration
2022-03-22T13:21:57.623Z [19211]: info: Processing controls parameters
2022-03-22T13:21:57.738Z [19211]: info: Successfully validated declaration
2022-03-22T13:21:57.801Z [19211]: info: Resolving parameters
2022-03-22T13:21:59.045Z [19211]: info: Executing install operations.
2022-03-22T13:21:59.056Z [19211]: info: Installing - do 1.27.0
2022-03-22T13:22:01.607Z [19211]: info: Validating - do extension is available.
2022-03-22T13:22:11.647Z [19211]: info: Installing - as3 3.34.0
2022-03-22T13:22:15.317Z [19211]: info: Validating - as3 extension is available.
2022-03-22T13:22:38.234Z [19211]: info: Installing - cf 1.10.0
2022-03-22T13:22:44.409Z [19211]: info: Validating - cf extension is available.
2022-03-22T13:22:57.477Z [19211]: info: Executing service operations.
2022-03-22T13:22:57.482Z [19211]: error: Unexpected token
 in JSON at position 780
2022-03-22T13:22:57.483Z [19211]: info: Sending F5 Teem report for failure case.
2022-03-22T13:22:58.172Z [19211]: info: {"id":"efb12381-01e5-c41a-78c0dccc25ef","product":"BIG-IP","cpuCount":8,"diskSize":81920,"memoryInMb":30160,"version":"16.1.0","nicCount":3,"regKey":"DIVTH-AGXAH-NWJPK-JIVCY-PFMWMHA","platformId":"Z100","hostname":"bigip1","management":"10.0.0.11/32","provisionedModules":{"ltm":"nominal"},"installedPackages":{"f5-service-discovery-1.10.6-1.noarch":"1.10.6","f5-declarative-onboarding-1.27.0-6.noarch":"1.27.0","f5-cloud-failover-1.10.0-0.noarch":"1.10.0","f5-appsvcs-3.34.0-4.noarch":"3.34.0"},"environment":{"pythonVersion":"Python 2.7.5","pythonVersionDetailed":"2.7.5 (default, Jun 22 2021, 22:47:51) \n[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]","nodeVersion":"v6.9.1","libraries":{"ssh":"OpenSSH_7.4p1, OpenSSL 1.0.2u-fips  20 Dec 2019"}}}
2022-03-22T13:23:10.182Z [19211]: info: F5 Teem report was successfully sent for failure case.
2022-03-22T13:23:10.182Z [19211]: info: Unexpected token
 in JSON at position 780

Not sure which JSON file is in play here. I will attach other log files to the issue also.

Note: admin password is not set either, and the devices are in Standalone / Active state at the end.  No configuration on the BIG-IPs - it is at the Setup utility once I reset admin password to login to TMUI
 
## Expected behavior
Expect it to create failover HA pair.
 
## Current behavior
BIG-IPs are in Standalone / Active with no configuration, including admin password.  

Yes, I have set password in Secrets Manager -- https://github.com/F5Networks/f5-google-gdm-templates-v2/issues/5.  
 
## Possible solution
Not sure -- will attach yaml file as well as log files.
 
## Steps to reproduce
1. Created YAML file - will attach.
2. Created mySecretId
 -- gcloud secrets create mySecretId --data-file="password.txt"
3.  Ran command to deploy
 --gcloud deployment-manager deployments create <deployment_name> --config <file.yaml>

## Screenshots 
![image](https://user-images.githubusercontent.com/2438670/159495659-84f89c63-3f2a-419a-a958-191489dabeac.png)

 
## Context
Just trying to get HA pair of BIG-IPs deployed - using sample YAML file with minimal changes
 
## Your Environment
 

 
- Version used:
- Environment name and version (e.g. Chrome 59, node.js 5.4, python 3.7.3):
- Operating System and version (desktop or mobile):
- Link to your project:

**Comment by curtkersey** [comment]
[bigIpRuntimeInit.log](https://github.com/F5Networks/f5-google-gdm-templates-v2/files/8324783/bigIpRuntimeInit.log)

**Comment by curtkersey** [comment]
[startup-script-post-swap-nic.log](https://github.com/F5Networks/f5-google-gdm-templates-v2/files/8324787/startup-script-post-swap-nic.log)

**Comment by curtkersey** [comment]
[cloud.cfg.txt](https://github.com/F5Networks/f5-google-gdm-templates-v2/files/8324790/cloud.cfg.txt)

**Comment by curtkersey** [comment]
[ck_failover.yaml.txt](https://github.com/F5Networks/f5-google-gdm-templates-v2/files/8324831/ck_failover.yaml.txt)

**Comment by curtkersey** [comment]
[runtime-init-conf.txt](https://github.com/F5Networks/f5-google-gdm-templates-v2/files/8326029/runtime-init-conf.txt)

**Comment by mikeshimkus** [comment]
That's odd, I get a 404 when I click on runtime-init-conf.txt.

Also it will help if you can send the contents of /var/log/restnoded/restnoded.log.

Feel free to contact me directly (in Teams, etc).

**Comment by curtkersey** [comment]
Thanks @mikeshimkus for your help!

For others, the problem was that when I set the BIG-IP password in mySecretId as described in https://github.com/F5Networks/f5-google-gdm-templates-v2/issues/5, the 'password.txt' file had a NEWLINE character at the end of the password.  I created the file with 'vi'.  The NEWLINE character in the secret was causing an error in the JSON file since that is an invalid character.

In order to set mySecretId without a NEWLINE character, I used the following command:
% echo -n "MyPassw0rd" | gcloud secrets versions add mySecretId  --data-file=-

After setting mySecretId like that, I re-ran the deployment, and it worked!
